src/pubsub/mod_pubsub_db_rdbms.erl

```diff
@@ -432,18 +432,24 @@ get_affiliation(Nidx, { LU, LS, _ }) ->
 -spec add_subscription(Nidx :: mod_pubsub:nodeIdx(),
                        LJID :: jid:ljid(),
                        Sub :: mod_pubsub:subscription(),
-                       SubId :: mod_pubsub:subId()) -> ok.
-add_subscription(Nidx, { LU, LS, LR }, Sub, SubId) ->
-    SQL = mod_pubsub_db_rdbms_sql:insert_subscription(Nidx, LU, LS, LR, sub2int(Sub), SubId),
+                       SubId :: mod_pubsub:subId(),
+                       SubOpts :: mod_pubsub:subOptions()) -> ok.
+add_subscription(Nidx, { LU, LS, LR }, Sub, SubId, SubOpts) ->
+    EncodedOpts = jsx:encode(SubOpts),
+    SQL = mod_pubsub_db_rdbms_sql:insert_subscription(Nidx, LU, LS, LR, sub2int(Sub),
+                                                      SubId, EncodedOpts),
     {updated, _} = mongoose_rdbms:sql_query(global, SQL),
     ok.
 
 -spec set_subscription_opts(Nidx :: mod_pubsub:nodeIdx(),
                             LJID :: jid:ljid(),
                             SubId :: mod_pubsub:subId(),
-                            Opts :: mod_pubsub:subOptions()) -> ok.
-set_subscription_opts(Nidx, LJID, SubId, Opts) ->
-    mod_pubsub_db_mnesia:set_subscription_opts(Nidx, LJID, SubId, Opts).
+                            Opts :: mod_pubsub:subOptions()) -> {result, []}.
+set_subscription_opts(Nidx, { LU, LS, LR }, SubId, Opts) ->
```

Comment:
According to the spec, the function returns `{result, []}` while in the code I can see it returns just `ok`.